### ISSUE TITLE
[Feature Request] Add Linux support

### ISSUE THREAD

**Issue by sudg6** [danielkrupinski/OneByteWallhack#34]
Could you point out what are the necessary adjustments should take place for the cheat to support Linux version of the game?

**Comment by mrjshzk** [comment]
Just compile your cheat to .so (shared object (pretty much the equivalent of dll in windows)) and instead of injecting client.dll, inject client.so

**Comment by sudg6** [comment]
> Just compile your cheat to .so (shared object (pretty much the equivalent of dll in windows)) and instead of injecting client.dll, inject client.so

you mean compiling the python file into a shared library file (.so)?

**Comment by mrjshzk** [comment]
No, your cheat. Compile it the same way you'd compile a DLL on windows, but a SO file.

**Comment by sudg6** [comment]
I might misunderstood the functionality of this source code. Is it an injector? I mean I must have .so file somewhere else then I use this python file to inject it?

**Comment by mrjshzk** [comment]
Nevermind, I just realized what this code does and you should not use it. You will get VAC banned.

This is pretty much the same thing as having cheat engine attached to csgo and assembly editing.

**Comment by mrjshzk** [comment]
And also I just checked and PyMem is a windows only library, so you would have to use another library compatible with linux (if it exists)

**Comment by sudg6** [comment]
> Nevermind, I just realized what this code does and you should not use it. You will get VAC banned.
> 
> This is pretty much the same thing as having cheat engine attached to csgo and assembly editing.

Which means it must be running on Windows! But where client.dll must be found?

**Comment by mrjshzk** [comment]
I don't want to just say it to you, I want you to learn from this, so follow along.
I went to the Pymem documentation and searched the 'module_from_name' function, which takes 2 arguments.

First argument is the 'process_handle' (int) and in this code that is 'pm.process_handle', being pm the csgo executable variable.

Now the second argument is 'module_name' (string) and in this code that is 'Client.dll'.

Also, the description of this function is the following 'Retrieve a module loaded by given process'.

So pretty much what this code does is ->
Pulls Client.dll from csgo process; edits the assembly code from said DLL and boom, done

I hope you have learned something :)

**Comment by sudg6** [comment]
Indeed, I did! Appreciated!
Since the package pymem is available on Windows only it's pointless to have it on Linux which I run as my main OS that's why I opened a request for Linux support in the first place.
________________________________
